tutorials/fiori-tools-mockserver-opa-testing/fiori-tools-mockserver-opa-testing.md

```diff
@@ -0,0 +1,606 @@
+    | `iChange...`        | change the content of a field                           |
+    | `iAdd/iRemove...`   | add/remove content to/from a list (e.g. filter-fields)  |
+    | `iPress...`         | click into a table-row or table-cell                    |
+    | etc.              |                                                         |
+
+
+    The basic principle of OPA test functions is to identify a UI object and then use it to carry out an action like pressing a button or checking content (assertion). Therefore, almost all functions of the API have an identifier as a first parameter which can either be an object containing parts of the stable ID of the UI control, or a string containing the label as a constant value.
+
+    Assertions often have a state parameter which allows you to check for any properties of a UI control. The coding sample below shows the assertion to check the visibility and enablement of a button. The identifier is passed as part of the service and action name.
+
+    ```Javascript
+    When.onTheDetailPage.onHeader()
+        .iCheckAction({ service: "com.c_salesordermanage_sd", action: "ChangeOrderStatus"},
+                      { visible: true, enabled: false });
+    ```
+
+5. Chaining of functions
+
+    When you call several functions that belong to the same UI area, the functions can be chained to ease readability. In the example below, the following assertions are executed on the object page:
+
+    - in the header the **Edit** and the **Change Order Status** buttons are checked
+    - with `.and.then.`, the control returns to the object page (`onTheDetailPage`) and a local test function `iSeeFacetActionButton()`, which is not part of the API, is called
+    - in the footer area, the **Save** button is checked
+
+    ```Javascript
+    When.onTheDetailPage
+      .onHeader()
+        .iCheckEdit({ visible: true, enabled: false })
+        .and.iCheckAction({ service: "com.c_salesordermanage_sd", action: "ChangeOrderStatus"},
+                          { visible: true, enabled: false })
+        .and.then.iSeeFacetActionButton("OrderData", "com.c_salesordermanage_sd.FacetFormAction",
+                                        { visible: true, enabled: true })
+        .and.onFooter()
+            .iCheckSave({ visible: true, enabled: true })
+    ```
+
+[DONE]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 5: ](Enhancing the OPA journey - testing the list report)]
+
+Steps 5 and 6 of this tutorial demonstrate the adoption of the SAP Fiori elements for OData V4 OPA test library.  
+
+The OPA test journey available in file `OpaJourney.js` is just a starting point for you to enhance the tests for the list report and the object page. This journey currently starts the application after the test data is restored. Then, the list report table is loaded and a function checks whether data available in the table. As a last step, the application is closed.
```

Comment:
"list report table" - Please use hyphens in compound words to improve readability. Otherwise it could mean that you want the user to "use list" and "report table". Please also change all other ocurrences.

Reply:
done